### PR TITLE
Crdcdh 1687 and 1693

### PR DESCRIPTION
**Pull request by vshand11**
Implemented CRDCDH 1687 and 1693 for adding and removing qcResult during validation and updating.

### CODE REVIEW

File: src/metadata_validator.py [hunk 1]
```diff
@@ -151,23 +152,35 @@ def validate(self, submission_id, scope):
...
             for record in data_records:
+                qc_result = get_qc_result(record, self.submission, VALIDATION_TYPE_METADATA, self.mongo_dao)
                 status, errors, warnings = self.validate_node(record)
-                
                 if errors and len(errors) > 0:
                     self.isError = True
-                    record[ERRORS] = errors
+                    # record[ERRORS] = errors
+                    qc_result[ERRORS] = errors
+                    qc_result["severity"] = "Error"
```
Comment (n2iw) [inline src/metadata_validator.py:165]: Please use appropriate constant for "Error"

File: src/metadata_validator.py [hunk 2]
```diff
@@ -151,23 +152,35 @@ def validate(self, submission_id, scope):
...
                 else:
-                    record[ERRORS] = []
+                    # record[ERRORS] = []
+                    qc_result[ERRORS] = []
                 if warnings and len(warnings)> 0: 
                     self.isWarning = True
-                    record[WARNINGS] = warnings
+                    # record[WARNINGS] = warnings
+                    qc_result[WARNINGS] = warnings
+                    qc_result["severity"] = "Warning"
```
Comment (n2iw) [inline src/metadata_validator.py:173]: Please use appropriate constant for "Warning"

**Review by n2iw — CHANGES_REQUESTED** [review]

File: src/metadata_validator.py (hunk 2 above)
Comment (vshand11) [reply src/metadata_validator.py]: Updated.  Please continue.

**Review by n2iw — CHANGES_REQUESTED** [review]
The **type** property for data file validation errors should be "data file", currently it's "file".

**Review by n2iw — APPROVED** [review]
Will merge and test.

**Pull request merged**